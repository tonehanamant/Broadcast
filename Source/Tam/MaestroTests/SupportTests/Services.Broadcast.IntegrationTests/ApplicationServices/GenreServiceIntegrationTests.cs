using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Services.Broadcast.ApplicationServices;
using Services.Broadcast.Entities.Enums;
using Tam.Maestro.Common.DataLayer;

namespace Services.Broadcast.IntegrationTests.ApplicationServices
{
    [TestFixture]
    [Category("short_running")]
    public class GenreServiceIntegrationTests
    {
        private const int MAESTRO_ID = (int)ProgramSourceEnum.Maestro;
        private const int REDBEE_ID = (int)ProgramSourceEnum.RedBee;
        private IGenreService _GenreService;

        [SetUp]
        public void Setup()
        {
            _GenreService  = IntegrationTestApplicationServiceFactory.GetApplicationService<IGenreService>();
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetAllMaestroGenres()
        {
            using (new TransactionScopeWrapper())
            {
                var genres = _GenreService.GetGenres(MAESTRO_ID);

                Approvals.Verify(IntegrationTestHelper.ConvertToJson(genres));
            }

        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetAllDativaGenres()
        {
            using (new TransactionScopeWrapper())
            {
                var genres = _GenreService.GetGenres(REDBEE_ID);

                Approvals.Verify(IntegrationTestHelper.ConvertToJson(genres));
            }

        }

    }
}
