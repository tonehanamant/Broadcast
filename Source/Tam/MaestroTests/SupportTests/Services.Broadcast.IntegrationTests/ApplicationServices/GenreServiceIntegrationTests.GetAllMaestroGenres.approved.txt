[
  {
    "Display": "Action",
    "Id": 1
  },
  {
    "Display": "Action/Adventure",
    "Id": 51
  },
  {
    "Display": "Adult",
    "Id": 2
  },
  {
    "Display": "Adventure",
    "Id": 3
  },
  {
    "Display": "Anthology",
    "Id": 4
  },
  {
    "Display": "Auction",
    "Id": 5
  },
  {
    "Display": "Awards",
    "Id": 6
  },
  {
    "Display": "Beauty & Fashion",
    "Id": 7
  },
  {
    "Display": "Business",
    "Id": 8
  },
  {
    "Display": "Children",
    "Id": 52
  },
  {
    "Display": "Comedy",
    "Id": 9
  },
  {
    "Display": "Competition",
    "Id": 10
  },
  {
    "Display": "Crime",
    "Id": 11
  },
  {
    "Display": "Do It Yourself",
    "Id": 13
  },
  {
    "Display": "Documentary",
    "Id": 12
  },
  {
    "Display": "Drama",
    "Id": 14
  },
  {
    "Display": "Educational",
    "Id": 53
  },
  {
    "Display": "Entertainment",
    "Id": 15
  },
  {
    "Display": "Family",
    "Id": 16
  },
  {
    "Display": "Fantasy",
    "Id": 17
  },
  {
    "Display": "Fitness & Exercise",
    "Id": 18
  },
  {
    "Display": "Food",
    "Id": 19
  },
  {
    "Display": "Game Show",
    "Id": 20
  },
  {
    "Display": "Health & Medicine",
    "Id": 21
  },
  {
    "Display": "History",
    "Id": 22
  },
  {
    "Display": "Holiday",
    "Id": 23
  },
  {
    "Display": "Home Shopping",
    "Id": 24
  },
  {
    "Display": "Horror",
    "Id": 25
  },
  {
    "Display": "Informational",
    "Id": 26
  },
  {
    "Display": "Inspirational",
    "Id": 27
  },
  {
    "Display": "Instructional",
    "Id": 50
  },
  {
    "Display": "Interview",
    "Id": 28
  },
  {
    "Display": "Investigative",
    "Id": 29
  },
  {
    "Display": "Lifestyle",
    "Id": 54
  },
  {
    "Display": "Music",
    "Id": 30
  },
  {
    "Display": "Musical",
    "Id": 31
  },
  {
    "Display": "Mystery",
    "Id": 32
  },
  {
    "Display": "Nature",
    "Id": 33
  },
  {
    "Display": "News",
    "Id": 34
  },
  {
    "Display": "Paid Program",
    "Id": 55
  },
  {
    "Display": "Parade",
    "Id": 35
  },
  {
    "Display": "Performance",
    "Id": 36
  },
  {
    "Display": "Politics",
    "Id": 37
  },
  {
    "Display": "Profile",
    "Id": 38
  },
  {
    "Display": "Reality",
    "Id": 39
  },
  {
    "Display": "Religious",
    "Id": 40
  },
  {
    "Display": "Romance",
    "Id": 41
  },
  {
    "Display": "Science & Technology",
    "Id": 43
  },
  {
    "Display": "Science Fiction",
    "Id": 42
  },
  {
    "Display": "Special",
    "Id": 56
  },
  {
    "Display": "Sports/Sports Talk",
    "Id": 44
  },
  {
    "Display": "Talk Show",
    "Id": 45
  },
  {
    "Display": "Telethon",
    "Id": 46
  },
  {
    "Display": "Thriller & Suspense",
    "Id": 47
  },
  {
    "Display": "Travel",
    "Id": 48
  },
  {
    "Display": "Western",
    "Id": 49
  }
]